src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -865,21 +872,21 @@ private bool SynchronizeBatch()
                 if (header.Height > tipToChase.Height)
                     break;
 
-                blockHashes.Add(header.HashBlock);
+                chainedHeadersSnapshot.Add(header.HashBlock, header);
 
                 if (++batchSize >= synchronizationBatchSize)
                     break;
             }
 
-            List<Block> blocks = this.blockRepository.GetBlocks(blockHashes);
+            List<Block> blocks = this.blockRepository.GetBlocks(chainedHeadersSnapshot.Select(c => c.Key).ToList());
```

Comment:
Generally we should not be calling `this.blockRepository` but we should get blocks via `ConsensusManager` so we look in cache first.

Probably not a fix for this PR.

Reply:
@dangershony you are right, I will make this change on his branch quick.

Reply:
@dangershony actually no, this change affects tests etc. @fenix2222 Please make this change in another PR?